p2p/src/net/mock/transport/tcp.rs

```diff
@@ -40,58 +42,75 @@ use crate::{
     P2pError, Result,
 };
 
+use self::encryption::{Encryption, NoiseEncryption};
+
 #[derive(Debug)]
-pub struct TcpMockTransport {}
+pub struct TcpMockTransportBase<E: Encryption>(std::marker::PhantomData<E>);
+
+// By default the transport uses Noise protocol encryption
+pub type TcpMockTransport = TcpMockTransportBase<NoiseEncryption>;
 
 #[async_trait]
-impl MockTransport for TcpMockTransport {
+impl<E: Encryption + 'static> MockTransport for TcpMockTransportBase<E> {
     type Address = SocketAddr;
     type BannableAddress = IpAddr;
-    type Listener = TcpListener;
-    type Stream = TcpMockStream;
+    type Listener = TcpMockListener<E>;
+    type Stream = TcpMockStream<E>;
 
     async fn bind(address: Self::Address) -> Result<Self::Listener> {
-        TcpListener::bind(address).await.map_err(Into::into)
+        let listener = TcpListener::bind(address).await?;
+        Ok(TcpMockListener(listener, Default::default()))
     }
 
     async fn connect(address: Self::Address) -> Result<Self::Stream> {
-        let stream = TcpStream::connect(address).await?;
-        Ok(TcpMockStream::new(stream))
+        let tcp_stream = TcpStream::connect(address).await?;
+        let noise_stream = TcpMockStream::new(tcp_stream, Side::Outbound).await?;
+        Ok(noise_stream)
     }
 }
 
+pub struct TcpMockListener<E: Encryption>(TcpListener, std::marker::PhantomData<E>);
```

Comment:
You don't need this trait-bound here

Reply:
Sure, removed. I just wanted to make it a little more readable.